config/v1/types_ingress.go

```diff
@@ -54,6 +54,22 @@ type IngressSpec struct {
 	// configurable routes.
 	// +optional
 	ComponentRoutes []ComponentRouteSpec `json:"componentRoutes,omitempty"`
+
+	// requiredHSTSPolicies specifies HSTS policies that are required to be set on
```

Comment:
question about the logic here.  I would have expected of the slice to matter, so evaluation would look like

1. check the route against each requiredHSTSPolicy index, in order
2. if the namespace selector and the domain pattern matches, the first  item in requiredHSTSPolicy is the one used
3. the accept/reject is based on that single item

Such an ordering concept will allow for punch-throughs by placing different policy early in the slice.

Reply:
Do you mean you'd like to see there be no requiredHSTSPolicy duplicated by index of namespace selector + domain pattern?

Reply:
Ok, yes, it does use that logic.  It is based on your WIP for the admission plugin.  I have a slightly revised WIP for the admission plugin here:  https://github.com/openshift/openshift-apiserver/pull/224/files#diff-cde26a88c8aa9772d3d1cbf96497ac0009504bc258fd662e2ad19b458dcd3d81
I'll change the comments here to make it clearer.

Reply:
Right, the plugin logic looks like what David describes:  Pick the first matching item (matching on domain and namespace label) and admit or reject based on that item.  That also seems like a good way to describe in this godoc how the list of policies is applied.  As it is, the description currently in the godoc is difficult to reason about because it has many cases and describes the matching logic in terms of matching the policy as well as the domain and the namespace label.

Reply:
I updated the comments, hope it is clarified.